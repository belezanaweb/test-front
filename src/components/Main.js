import React from 'react';
import { Switch, Route } from 'react-router';
import Cart from '../containers/Cart';
import Payment from '../containers/Payment';
import Confirmation from '../containers/Confirmation';

const Main = () => (
    <main>
        <Switch>
            <Route exact path='/' component={Cart} />
            <Route path='/payment' component={Payment} />
            <Route path='/confirmation' component={Confirmation} />
        </Switch>
    </main>
)

export default Main;