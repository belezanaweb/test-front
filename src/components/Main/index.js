import React, { PureComponent, Fragment } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import { getProducts } from '../../store/ducks/cart'

class Main extends PureComponent {
  async componentDidMount () {
    await this.props.getProducts()
  }

  render () {
    const { children } = this.props

    return <Fragment>{children}</Fragment>
  }
}

Main.propTypes = {
  getProducts: PropTypes.func.isRequired,
  children: PropTypes.arrayOf(PropTypes.node).isRequired
}

const mapDispatchToProps = dispatch => ({
  getProducts: () => dispatch(getProducts())
})

export default connect(
  null,
  mapDispatchToProps
)(Main)
