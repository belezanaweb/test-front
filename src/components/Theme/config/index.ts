export { default as light } from "./light";
