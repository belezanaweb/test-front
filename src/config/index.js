export const cartSteps = ['Sacola', 'Pagamento', 'Confirmação']
