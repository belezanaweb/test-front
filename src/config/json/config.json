{
    "ENDPOINT_BASKET": "https://www.mocky.io/v2/5b15c4923100004a006f3c07"
}