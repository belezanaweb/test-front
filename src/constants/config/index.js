export const BASE_URL = 'http://www.mocky.io/v2/'
export const BAG_ID = '5b15c4923100004a006f3c07'
