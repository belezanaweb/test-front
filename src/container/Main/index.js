import { Grid } from '@mui/material'
import React from 'react'
import { MainContainer } from './style.js'
import Cart from '../Cart'
import Payment from '../Payment'
import Confirmation from '../Confirmation'
import { Header } from '../../components/index.js'

const Main = () => {
  return (
    <>
      <MainContainer>
        <Header typeTab={'SACOLA'} />
        <Grid container alignContent="center" style={{ marginTop: '14%' }}>
          <Grid item xs={12}>
            <Payment />
          </Grid>
        </Grid>
      </MainContainer>
    </>
  )
}

export default Main
