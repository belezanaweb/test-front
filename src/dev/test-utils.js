// source: https://testing-library.com/docs/react-testing-library/setup#custom-render
import React, { Suspense } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'emotion-theming'
import theme from '../theme'

// eslint-disable-next-line react/prop-types
function WithProviders ({ children }) {
  return (
    <ThemeProvider theme={theme}>
      <Suspense fallback={<>Loading...</>}>{children}</Suspense>
    </ThemeProvider>
  )
}

const customRender = (ui, options) => render(ui, { wrapper: WithProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
