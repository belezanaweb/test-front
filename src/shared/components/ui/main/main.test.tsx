import React from 'react';
import { render, screen } from '../../../../../config/test-utils';

import Main from './index';

describe('<Main />', () => {
  it('should render component with <main> tag', () => {
    render(<Main>Main</Main>, {});
    expect(screen.getByRole('main'));
  })

  it('should render component with default props and match snapshot', () => {
    const { container } = render(<Main>Main</Main>, {});
    expect(container.firstChild).toMatchSnapshot();
  })
})
