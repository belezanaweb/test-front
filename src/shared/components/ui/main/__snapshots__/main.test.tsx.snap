// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render component with default props and match snapshot 1`] = `
.c0 {
  padding: 1rem;
}

<main
  class="c0"
>
  Main
</main>
`;
